galicaster/utils/ical.py

```diff
@@ -62,13 +62,44 @@ def get_updated_events(old_events, new_events):
     out = list()
     for old_event in old_events:
         for new_event in new_events:
-            if (old_event['UID'] == new_event['UID'] and
-                old_event['DTSTART'].dt.replace(tzinfo=None) != new_event['DTSTART'].dt.replace(tzinfo=None)):
-                out.append(old_event)
+            append = False
+            if old_event['UID'] == new_event['UID']:
+                if (old_event['DTSTART'].dt.replace(tzinfo=None) != new_event['DTSTART'].dt.replace(tzinfo=None) or
+                    old_event['DTEND'].dt.replace(tzinfo=None) != new_event['DTEND'].dt.replace(tzinfo=None) or
+                    old_event['SUMMARY'] != new_event['SUMMARY']):
+                    append = True
+                if 'RELATED-TO' in new_event and not 'RELATED-TO' in old_event or not 'RELATED-TO' in new_event and 'RELATED-TO' in old_event:
+                    append = True
+                elif 'RELATED-TO' in new_event and 'RELATED-TO' in old_event and old_event['RELATED-TO'] != new_event['RELATED-TO']:
+                    append = True
+                if 'ORGANIZER' in new_event and not 'ORGANIZER' in old_event or not 'ORGANIZER' in new_event and 'ORGANIZER' in old_event:
+                    append = True
+                elif 'ORGANIZER' in new_event and 'ORGANIZER' in old_event and old_event['ORGANIZER'] != new_event['ORGANIZER']:
+                    append = True
+                for attach_enc in new_event['ATTACH']:
+                    attach =  base64.b64decode(attach_enc)
+                    if not attach_enc.params['X-APPLE-FILENAME'] == 'episode.xml' and not attach_enc.params['X-APPLE-FILENAME'] == 'series.xml':
+                        break
+                for attach_enc in old_event['ATTACH']:
+                    attach1 =  base64.b64decode(attach_enc)
+                    if not attach_enc.params['X-APPLE-FILENAME'] == 'episode.xml' and not attach_enc.params['X-APPLE-FILENAME'] == 'series.xml':
+                        break
+
+                attach_split = attach.split("\n",2)
+                attach = attach_split[0] + attach_split[2]
+
+                attach_split = attach1.split("\n",2)
+                attach1 = attach_split[0] + attach_split[2]
+
+                if attach != attach1:
+                    append = True
+
+                if append:
+                    out.append(new_event)
     return out
 
 
-def create_mp(repo, event):
+def create_mp(repo, event, rewrite1=None):
```

Comment:
I think the name `rewrite1` is not clear enough. It should be called something more close to what it does.
Judging by the code below, maybe `override_rewrite`? Or maybe call it `rewrite` and check below if it's defined before changing the value:
```python
if rewrite == None:
    rewrite = True
```